chasm/engine.go

```diff
@@ -234,18 +241,18 @@ func ReadComponent[C Component, R []byte | ComponentRef, I any, O any](
 	return output, err
 }
 
-type PollComponentRequest[C Component, I any, O any] struct {
-	Ref         ComponentRef
-	PredicateFn func(C, Context, I) bool
-	OperationFn func(C, MutableContext, I) (O, error)
-	Input       I
-}
-
-func PollComponent[C Component, R []byte | ComponentRef, I any, O any, T any](
+// PollComponent waits until the predicate is true when evaluated against the component identified
+// by the supplied component reference. If this times out due to a server-imposed long-poll timeout
+// then it returns (nil, nil, nil), as an indication that the caller should continue long-polling.
+// Otherwise it returns (output, ref, err), where output is the output of the predicate function,
+// and ref is a component reference identifying the state at which the predicate was satisfied. The
+// predicate must be monotonic: if it returns true at execution state transition s then it must
+// return true at all transitions t > s. If the predicate is true at the outset then PollComponent
+// returns immediately. opts are currently ignored.
+func PollComponent[C Component, R []byte | ComponentRef, I any, O any](
 	ctx context.Context,
 	r R,
-	predicateFn func(C, Context, I) (T, bool, error),
-	operationFn func(C, MutableContext, I, T) (O, error),
+	monotonicPredicate func(C, Context, I) (O, bool, error),
 	input I,
 	opts ...TransitionOption,
```

Comment:
We can get rid of the transition options right?

Reply:
We can. `ReadComponent` and `UpdateComponent` also have unused `opts`. So let's remove them in a subsequent PR (if that's what @yycptt wants).